src/box/vy_mem.c

```diff
@@ -126,6 +130,8 @@ void
 vy_mem_delete(struct vy_mem *index)
 {
 	index->env->tree_extent_size -= index->tree_extent_size;
+	/* TODO: Destroy the index->tree? */
+	matras_allocator_destroy(&index->matras_allocator);
```

Comment:
Shouldn't we destroy the mem tree here?

Reply:
There's no need, because the `lsregion` allocator doesn't support freeing memory blocks. Destroying the tree would result in unnecessary work.

Reply:
So should I drop the matras allocator deinitialization too? It only frees the reserved blocks.

Reply:
I guess so. Let's add a comment explaining why we don't call them.

Reply:
Fixed.

```diff
--- a/src/box/vy_mem.c
+++ b/src/box/vy_mem.c
@@ -130,8 +130,13 @@ void
 vy_mem_delete(struct vy_mem *index)
 {
        index->env->tree_extent_size -= index->tree_extent_size;
-       /* TODO: Destroy the index->tree? */
-       matras_allocator_destroy(&index->matras_allocator);
+       /*
+        * It would be expected to destroy the index->matras_allocator and
+        * the index->tree here, but they simply free allocated and reserved
+        * memory blocks, and the lsregion used for allocation of the memory
+        * does not support freeing arbitrary blocks, so they're effectively
+        * no-ops. Let's omit them.
+        */
        tuple_format_unref(index->format);
        fiber_cond_destroy(&index->pin_cond);
        TRASH(index);
```

Reply:
Fixed.

```diff
--- a/src/box/vy_mem.c
+++ b/src/box/vy_mem.c
@@ -130,8 +130,13 @@ void
 vy_mem_delete(struct vy_mem *index)
 {
        index->env->tree_extent_size -= index->tree_extent_size;
-       /* TODO: Destroy the index->tree? */
-       matras_allocator_destroy(&index->matras_allocator);
+       /*
+        * It would be expected to destroy the index->matras_allocator and
+        * the index->tree here, but they simply free allocated and reserved
+        * memory blocks, and the lsregion used for allocation of the memory
+        * does not support freeing arbitrary blocks, so they're effectively
+        * no-ops. Let's omit them.
+        */
        tuple_format_unref(index->format);
        fiber_cond_destroy(&index->pin_cond);
        TRASH(index);
```